### DISCUSSION TITLE
IK for humanoid robot

### DISCUSSION THREAD

**Discussion by eirtech** [stack-of-tasks/pinocchio#2118] [General]
Hi, 
Thanks again for your nice work.

I want to calculate IK for humanoid robot. I want to consider for example hip joint as a fix joint (or any other joint as I want) and then calculate IK. I wanted to know how could I define it in Pinocchio? 

Thanks in advance.

**Comment by stephane-caron** [comment]
Pinocchio provides all the building blocks to implement inverse kinematics. You can look for instance at [Pink](https://github.com/stephane-caron/pink/) which implements a weighted-tasks approach with quadratic programming (NB: I'm the original developer and a contributor to that project). Pink has two (similar) humanoid examples:

- [`humanoid_draco3.py`](https://github.com/stephane-caron/pink/blob/06581551e2eae16de9571c73c28005101f2f352f/examples/humanoid_draco3.py`) for the DRACO 3 humanoid:

https://github.com/stephane-caron/pink/assets/1189580/db6acda8-82a4-4f4d-9acf-1fc3d831e222

- [`humanoid_jvrc.py`](https://github.com/stephane-caron/pink/blob/06581551e2eae16de9571c73c28005101f2f352f/examples/humanoid_jvrc.py) for the JVRC-1 humanoid.

You can use Pink as a library, or look into its code to see how things are done and implement your own IK. Hoping this helps!

**Reply by eirtech** [reply]
Thanks for your quick response. It would of course help.
